src/evaluate.py

```diff
@@ -203,7 +202,7 @@ def write_preds(
             _write_winograd_preds(
                 task, preds_df, pred_dir, split_name, strict_glue_format=strict_glue_format
             )
-        elif isinstance(task, GLUEDiagnosticTask):
+        elif task.is_diagnostic:
```

Comment:
I'd vote to revert this change. For deciding whether to train, it makes sense to check a simple Boolean that we can easily reuse for new tasks. But for deciding how to write preds, we actually need to know something about tho data structures used for the task, so I think it's right to check isinstance or something similarly specific.